src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -319,6 +319,18 @@ private async Task StartContainerAsync(IExecutionContext executionContext, Conta
             {
                 // Ensure .taskkey file exist so we can mount it.
                 string taskKeyFile = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Work), ".taskkey");
+                // If the previous clean up failed, we want to start with an empty taskkey file so try deleting again
```

Comment:
Why do we need this?

Reply:
I meant to call this out, this should not have been in this PR. I forgot to revert this as I was working on a potential solution for an issue but I talked myself out of this being the right approach. However, as I type this, I am waffling as to whether this should be left in. We can discuss tomorrow.

Reply:
Cool, this is the only potential uncertainty I have with this PR, other than that LGTM